### PR TITLE
CAppParamParser: add new tty argument for gbm windowing

### PR DESCRIPTION
**Pull request by lrusak**
Since no one had any strong feelings when I asked yesterday I figured I would PR this just to get some more exposure.

This allows us to specify the tty to start kodi on. This is especially useful for starting kodi-gbm from an ssh session. This **does not** allow kodi to change VT's automatically as that would require `sudo` rights. So you still have to select a valid VT for this to work.

```
kodi -fs --standalone --tty=/dev/tty1
```

This is all done in similar fashion to how weston does it upstream.
see, https://gitlab.freedesktop.org/wayland/weston/blob/master/libweston/weston-launch.c#L525-588

### CODE REVIEW

**Comment by lrusak** [comment]
Anyone have any issues with this?

**Comment by a1rwulf** [comment]
+1 because I also do ssh into my devbox.

File: xbmc/AppParamParser.cpp [hunk 1]
```diff
@@ -62,6 +62,7 @@ void CAppParamParser::DisplayHelp()
   printf("  --test\t\tEnable test mode. [FILE] required.\n");
   printf("  --settings=<filename>\t\tLoads specified file after advancedsettings.xml replacing any settings specified\n");
   printf("  \t\t\t\tspecified file must exist in special://xbmc/system/\n");
+  printf("  --tty=<tty>\t\tStart session on alternative tty. e.g. --tty=/dev/tty4 (GBM windowing only)\n");
```
Comment (Rechi) [inline xbmc/AppParamParser.cpp]: IMO, this should be only show if building for GBM.

**Comment by pkerling** [comment]
Mh looking at this it kind of screams "refactor the cmdline argument parsing and give modules the opportunity to hook into it" ... Not sure if you want to do it for tihs PR though.

**Comment by lrusak** [comment]
>we could get around including CApp by using an env var. I.e. appparamparser could set KODI_TTY, if not already set. That would have the added benefit of also working with just the env var set.

yes I guess I could just do `getenv(KODI_TTY)` or something. Not sure if that's a better solution.

>Mh looking at this it kind of screams "refactor the cmdline argument parsing and give modules the opportunity to hook into it" ... Not sure if you want to do it for tihs PR though.

Yes, that's why I said it may be better to register it with `CServiceBroker` and allow grabbing parameters from there.

**Comment by wsnipex** [comment]
on a related note: even with this PR, changes are needed to start kodi-gbm via systemd service:
https://forum.kodi.tv/showthread.php?tid=336825&pid=2785722#pid2785722

File: xbmc/AppParamParser.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/AppParamParser.cpp]: updated

**Comment by lrusak** [comment]
closing due to #14776

**Pull request closed without merging**